lib/webdriveragent.js

```diff
@@ -149,6 +150,32 @@ class WebDriverAgent {
 
     await B.all(stops);
   }
+
+  //Body must be object
+  async sendCommandWithSession(command, body, method) {
```

Comment:
What about the request are you trying to modify?
What we can do is implement a custom proxy. right now we use the default that is imported from `appium-jsonwp-proxy`, and we use it here: https://github.com/appium/appium-xcuitest-driver/blob/master/lib/webdriveragent.js#L103

It does the sessionId translation, and also changes the host and port.

Here's all the logic it handles: https://github.com/appium/jsonwp-proxy/blob/master/lib/proxy.js
There's lots of stuff in there to make sure we keep everything proxied nicely, and report errors nicely too.

We could modify it or add a custom function that transposes the packets, what change needs to be made to requests?